ci/scripts/integration_arrow.sh

```diff
@@ -24,9 +24,16 @@ source_dir=${1}/cpp
 build_dir=${2}/cpp
 gold_dir_0_14_1=$arrow_dir/testing/data/arrow-ipc-stream/integration/0.14.1
 gold_dir_0_17_1=$arrow_dir/testing/data/arrow-ipc-stream/integration/0.17.1
+gold_dir_1_0_0_le=$arrow_dir/testing/data/arrow-ipc-stream/integration/1.0.0-littleendian
+gold_dir_1_0_0_be=$arrow_dir/testing/data/arrow-ipc-stream/integration/1.0.0-bigendian
 
 pip install -e $arrow_dir/dev/archery
 
 archery integration --with-all --run-flight \
     --gold-dirs=$gold_dir_0_14_1 \
     --gold-dirs=$gold_dir_0_17_1 \
+
+# TODO: support other languages
+archery integration --with-cpp=1 --run-flight \
```

Comment:
Since other languages (e.g. Java) cause failure of integration tests, future separate PR will fix failures in other languages.   
This PR focuses on cpp integration test.

Reply:
You didn't answer my question above :-)

Also, instead of running `archery integration` a second time, you should add the new "gold dirs" to the command above, and add selective skips like in https://github.com/apache/arrow/blob/master/dev/archery/archery/integration/runner.py#L129